app/src/main/java/com/example/next_show/adapters/ShowAdapter.java

```diff
@@ -84,16 +86,19 @@ public void onClick(View v) {
             if (position != RecyclerView.NO_POSITION) {
                 Show show = shows.get(position);
 
-                // make new instance of the fragment
-                ShowDetailFragment fragment = ShowDetailFragment.newInstance(show, FROM_SAVED_FRAGMENT /* false */);
+                // create new bundle of arguments to pass into Navigation
+                Bundle bundle = new Bundle();
+                bundle.putParcelable("Show", show);
+                bundle.putBoolean("savedBool", fromSavedFragment);
 
-                // launch the fragment and commit
-                AppCompatActivity currActivity = (AppCompatActivity) context;
-                FragmentTransaction detailTransact = currActivity.getSupportFragmentManager().beginTransaction();
-                detailTransact.replace(R.id.fragment_container_view, fragment);
-                detailTransact.commit();
+                if(fromSavedFragment){
+                    // from saved to details
+                    Navigation.findNavController(v).navigate(R.id.action_savedFragment_to_showDetailFragment, bundle);
+                    return;
+                }
 
-                // TODO: navigate back to original feed without reloading all of that information again
+                // from feed to details
+                Navigation.findNavController(v).navigate(R.id.action_feedFragment_to_showDetailFragment, bundle);
```

Comment:
Yeah, we could pass something in that colors this:

```
interface SomeNavInterface {
   fun navigate(view, postBundle)
}

interface NavigateToSavedShows : SomeNavInterface {
  fun navigate(view, bundle) {
    Navigation.findNavController(v).navigate(R.id.action_savedFragment_to_showDetailFragment, bundle);
  }
}
```
...

And then for your view holder, you just would call:
```
  navigator.navigate(v, postBundle);
```